core/src/main/java/org/apache/iceberg/ManifestFiles.java

```diff
@@ -167,6 +167,29 @@ public static ManifestWriter<DataFile> write(
         "Cannot write manifest for table version: " + formatVersion);
   }
 
+  /**
+   * Create a new {@link RollingManifestWriter} for the given format version.
+   *
+   * @param formatVersion a target format version
+   * @param snapshotId a snapshot ID for the manifest entries, or null for an inherited ID
+   * @param spec a {@link PartitionSpec}
+   * @param io a {@link FileIO}
+   * @param outputFileFactory a {@link ManifestOutputFileFactory} to generate the manifest output
+   *     file
+   * @param targetFileSizeInBytes the target file size for manifest files
+   * @return a rolling manifest writer which could generate multiple manifest files
+   */
+  public static RollingManifestWriter<DataFile> rollingWrite(
```

Comment:
Hm, the method is public but the exposed class is package-private. Let me think.

Reply:
Are we sure this is the best way to expose rolling writers in general? Have we considered a single parameterized rolling writer that accepts a factory of manifest writers?

Reply:
Let me think, no need to update anything now.